lib/models/repository-states/present.js

```diff
@@ -196,7 +196,10 @@ export default class Present extends State {
 
       if (filePathEndsWith(event.path, '.git', 'config')) {
         // this won't catch changes made to the template file itself...
-        const template = await this.fetchCommitMessageTemplate();
```

Comment:
Great! Thanks for tackling this @annthurium! 

I have a readability/logic suggestion -- in the case where there is no template we can update the commit message directly and null out our cached `commitMessageTemplate`. This way, we use the term `template` exclusively for what would be provided by Git as the commit message template and not artificially making it an empty string. And we maintain the current cohesion between `this.commitMessageTemplate` and the return value of `fetchCommitMessageTemplate()` (they are both `null` in the case of no template configuration).

```
const template = await this.fetchCommitMessageTemplate();
if (template === null) {
  this.setCommitMessage('');
  this.setCommitMessageTemplate(null);
} else if (this.commitMessageTemplate !== template) {
  this.setCommitMessageTemplate(template);
  this.setCommitMessage(template);
}
```

Reply:
cool, thanks for the suggestion.

since the commit message template is set in either case, I'll do that outside the if statements.  Other than that, did it the way you suggest.